mirrord/layer/src/debugger_ports.rs

```diff
@@ -91,81 +99,142 @@ impl FromStr for DebuggerType {
+                .into_iter()
+                .collect::<Vec<_>>()
             }
             Self::JavaAgent => {
-                let is_java = args.first()?.ends_with("java");
-
-                if !is_java {
-                    None?
+                let is_java = args.first().unwrap_or(&"".to_string()).ends_with("java");
+
+                if is_java {
+                    args.iter()
+                        .find_map(|arg| arg.strip_prefix("-agentlib:jdwp=transport=dt_socket"))
+                        .and_then(|agent_lib_args| {
+                            agent_lib_args
+                                .split(',')
+                                .find_map(|arg| arg.strip_prefix("address="))
+                        })
+                        .and_then(|full_address| full_address.split(':').last())
+                        .and_then(|port| port.parse::<u16>().ok())
+                        .map(|port| SocketAddr::new(IpAddr::V4(Ipv4Addr::LOCALHOST), port))
+                } else {
+                    None
                 }
-
-                args.iter()
-                    .find_map(|arg| arg.strip_prefix("-agentlib:jdwp=transport=dt_socket"))
-                    .and_then(|agent_lib_args| agent_lib_args.split(',').find_map(|arg| arg.strip_prefix("address=")))
-                    .and_then(|full_address| full_address.split(':').last())
-                    .and_then(|port| port.parse::<u16>().ok())
-                    .map(|port| SocketAddr::new(IpAddr::V4(Ipv4Addr::LOCALHOST), port))
-
+                .into_iter()
+                .collect::<Vec<_>>()
             }
-        }
-        .and_then(|addr| match addr.ip() {
+            Self::NodeInspector => {
+                let is_node = args.first().unwrap_or(&"".to_string()).ends_with("node");
+
+                if is_node && let Ok(value) = std::env::var("NODE_OPTIONS") {
+                    // matching specific flags so we avoid matching on, for example,
+                    // `--inspect-publish-uid=http`
+                    let flags = value.split(" --").filter(|s| {
+                        s.contains("inspect=")
+                            || s.contains("inspect-brk=")
+                            || s.contains("inspect-wait=")
+                    });
+                    flags
+                        .filter_map(|flag| {
+                            let vec = flag.split('=').collect::<Vec<_>>();
+                            match vec.as_slice() {
+                                // you can use --inspect, -wait and -brk all together at once -
+                                // we need to ignore them all
+                                &[_flag, port] => port.parse::<u16>().ok(),
+                                _ => None,
+                            }
+                        })
+                        .map(|port| SocketAddr::new(IpAddr::V4(Ipv4Addr::LOCALHOST), port))
+                        .collect::<Vec<SocketAddr>>()
```

Comment:
wdyt about this:
```rust
    var.split_ascii_whitespace()
        .filter_map(|flag| match flag.split_once('=')? {
            ("--inspect" | "--inspect-brk" | "--inspect-wait", port) => port.parse::<u16>().ok(),
            _ => None,
        })
        .map(|port| SocketAddr::new(Ipv4Addr::LOCALHOST.into(), port))
        .collect::<Vec<_>>()
```

Reply:
Also, can these flags use default values? Like if you encounter just `--inspect`, does it work and use a default port?

Reply:
i think the reason i didnt split on whitespace is because path names can contain whitespace but this is better. As far as defaults go, if you invoke with `--inspect`, the flag gets transformed by node to `--inspect=xyz` with the port number :)